### DISCUSSION TITLE
Class derived from BlockGroup breaks ImageGallery in the Manager

### DISCUSSION THREAD

**Discussion by debeardjr** [PiranhaCMS/piranha.core#1628] [Q&A] [answered]
Hello, I have just started learning Piranha CMS and I have to say, I am very impressed!

While learning the features and abilities of Piranha, I created a simple class that inherits BlockGroup and it works great. However, now the ImageGallery Block does not work in the Manager. When I click on the Add button, nothing happens. If I do not register my custom BlockGroup, the ImageGallery will work. Here is the code from the custom BlockGroup class:
`[BlockGroupType(Name = "PDF List", Category = "Media", Icon = "far fa-file-pdf")]
[BlockItemType(Type = typeof(HtmlBlock))]
public class PdfGalleryBlock : BlockGroup`
 {
      public StringField Title { get; set; }
 }`
![image](https://user-images.githubusercontent.com/12800061/118052879-89c45380-b351-11eb-9a89-1e33867ab9f1.png)

Is there something incorrect in the way I have set up the PdfGalleryBlock class?

**Comment by debeardjr** [comment, marked as answer]
Figured this out. The Piranha Page class was decorated with multiple [BlockItemType] attributes and was missing one for HtmlBlock, which the BlockGroup depends on.

**Reply by tidyui** [reply]
Well maybe this is not that logical in a sense. If you explicitly add a group type you probably expect that the items available for that group should be included in the group context, but not on “root level”. At the moment you can only allow/disallow block types in the whole scope which of course means that if you allow the HtmlBlock editors will be able to add it outside of the group as well.

**Reply by debeardjr** [reply]
Thank you for the feedback. I was able to resolve this by creating two BlockTypes. The first is a Block (named DocumentBlock) that is declared with the "IsUnlisted" Property on the BlockType Attribute. 
The second is a BlockGroup with BlockItemType of DocumentBlock. 

This ultimately prevents users from seeing the DocumentBlock in the Editor and works perfectly.

```c#
    // DocumentBlock Class
    [BlockType(Name = "Document", IsUnlisted = true)]
    public class DocumentBlock : Block
    {
    ...
    }

    // DocumentGalleryBlock Class
    [BlockGroupType(Name = "Document Gallery", Category = "Media", Icon = "far fa-copy")]
    [BlockItemType(Type = typeof(DocumentBlock))]
    public class DocumentGalleryBlock : BlockGroup
    {
    ...
    }
